homeassistant/components/samsungtv/media_player.py

```diff
@@ -136,32 +138,55 @@ def __init__(self, host, port, name, timeout, mac, uuid):
             "name": "HomeAssistant",
             "description": name,
             "id": "ha.component.samsung",
+            "method": None,
             "port": port,
             "host": host,
             "timeout": timeout,
         }
 
+        # Select method by port number, mainly for fallback
         if self._config["port"] in (8001, 8002):
             self._config["method"] = "websocket"
-        else:
+        elif self._config["port"] == 55000:
             self._config["method"] = "legacy"
 
-    def update(self):
+    async def async_update(self):
```

Comment:
We're not allowed to do sync I/O in async context, ie inside the event loop. `self.send_key` will do I/O by the looks of it, so we can't change here to a coroutine.

Reply:
When I do this some tests are hanging.

Reply:
What tests are hanging and what are we doing in those tests?

Reply:
Had forgotten the `remote` fixture on 3 tests. No it's working.